homeassistant/components/lacrosse_view/sensor.py

```diff
@@ -44,9 +45,28 @@ class LaCrosseSensorEntityDescription(
     """Description for LaCrosse View sensor."""
 
 
-def get_value(sensor: Sensor, field: str) -> float:
+def get_value(sensor: Sensor, field: str) -> float | int | str:
     """Get the value of a sensor field."""
-    return float(sensor.data[field]["values"][-1]["s"])
+    value = sensor.data[field]["values"][-1]["s"]
+    # Check if it is already a float or int
+    if isinstance(value, int):
+        return value
+    if isinstance(value, float):
+        return value
+
+    if (value.find("-") <= 0) and value.replace(
+        "-", "", 1
+    ).isdigit():  # Check if int, then return int
+        return int(value)
+    if (
+        (value.find("-") <= 0)
+        and (value.count(".") < 2)
+        and (value.replace("-", "", 1).replace(".", "", 1).isdigit())
+    ):  # Check if float, then return float
+        return float(value)
+
+    # Return string value of field
+    return str(value)
```

Comment:
Assuming there *is* a compelling reason for an `int`, I'd strongly discourage this particular method of parsing. Try instead:
```suggestion
    value = sensor.data[field]["values"][-1]["s"]
    try:
        value = float(value)
    except ValueError:
        return value
    return int(value) if value.is_integer() else value
```